### PR TITLE
doc: add failures guide

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony @joshix @gyuho @philips @robszumski 

fix #https://github.com/coreos/etcd/issues/3428

### CODE REVIEW

**Comment by gyuho** [comment]
lgtm.

File: Documentation/op-guide/failures.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Understand failures
+
+Failures are common in a large deployment of machines. A machine fails when its hardware or software malfunctions. Multiple machines fail together when there are power failures or network issues. Multiple kinds of failures can also happen at once; it is almost impossible to enumerate all possible failure cases. 
+
+In this section, we catalog kinds of failures and discuss how etcd is designed to tolerant these failures. Most users, if not all, can map a particular failure into one kind of failure. To prepare for rare or [unrecoverable failures][unrecoverable], always [back up][backup] the etcd cluster.
```
Comment (heyitsanthony) [inline Documentation/op-guide/failures.md]: s/tolerant/tolerate

File: Documentation/op-guide/failures.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+A network partition is similar to a minor followers failure or a leader failure. A network partition divides the etcd cluster into two parts; one with a member majority and the other with a member minority. The majority side becomes the available cluster and the minority side is unavailable; there is no “split-brain” in etcd. 
+
+If the leader is on the majority side, then from the majority point of view the failure is a minority follower failure. If the leader is on the minority side, then it is a leader failure. The leader on the minority side steps down and the majority side elects a new leader.
+
+Once the network partition clears, the minority side automatically recognizes the leader from the majority side and recovers its state. 
+
+## Failure during bootstrapping
+
+A cluster bootstrap is only successful if all required members successfully start. If there any failure happens during bootstrapping, remove the data directories on all members and re-bootstrap the cluster with a new cluster-token or new discovery token.
```
Comment (heyitsanthony) [inline Documentation/op-guide/failures.md]: s/If there/If any/

**Comment by heyitsanthony** [comment]
lgtm after correcting nits

**Comment by joshix** [comment]
Gentlemen, this is beautiful, and I very personally thank you for it.

**Pull request merged**